File: Python/String - 2/xyz_there.py
def xyz_there(str):
    
    if 'xyz' not in str:
        
        return False
    
    if str[:3] == 'xyz':
        
        return True
    
    for i in range(len(str)):
        
        if str[i]!='.' and str[i+1:i+4] =='xyz':
            
            return True
        
    return False

File: Java/String - 2/xyzThere.java
public boolean xyzThere(String str) {

    if(str.length()<3) return false;

    if(str.indexOf("xyz")==0) return true;

    for(int i = 1; i<= str.length()-3;i++) {

        String temp = str.substring(i,i+3);

        if(temp.equals("xyz") && str.charAt(i-1) !='.')

        return true;

    }

    return false;

}